integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/apim/APIMWithMgwBaseTest.java

```diff
@@ -0,0 +1,336 @@
+        } catch (InterruptedException ignored) {
+
+        }
+    }
+
+    /**
+     * This method can be used to wait for API deployment sync in distributed and clustered environment APIStatusMonitor
+     * will be invoked to get API related data and then verify that data matches with expected response provided.
+     *
+     * @param apiProvider      - Provider of the API
+     * @param apiName          - API name
+     * @param apiVersion       - API version
+     * @param expectedResponse - Expected response
+     * @throws MicroGWTestException if something goes wrong when getting the tenant identifier
+     */
+    void waitForAPIDeploymentSync(String apiProvider, String apiName, String apiVersion,
+                                  String expectedResponse)
+            throws XPathExpressionException, MicroGWTestException {
+
+        long currentTime = System.currentTimeMillis();
+        long waitTime = currentTime + (60 * 1000);
+
+        String colonSeparatedHeader =
+                apimServerContext.getContextTenant().getTenantAdmin().getUserName() + ":" + apimServerContext
+                        .getContextTenant().getTenantAdmin().getPassword();
+
+        String authorizationHeader = "Basic " + new String(Base64.encodeBase64(colonSeparatedHeader.getBytes()));
+        Map<String, String> headerMap = new HashMap();
+        headerMap.put("Authorization", authorizationHeader);
+        String tenantIdentifier = getTenantIdentifier(apiProvider);
+
+        while (waitTime > System.currentTimeMillis()) {
+            HttpResponse response = null;
+            try {
+                response = HttpRequestUtil.doGet(
+                        apimServiceURLHttps + "APIStatusMonitor/apiInformation/api/" + tenantIdentifier + apiName + "/"
+                                + apiVersion, headerMap);
+            } catch (IOException ignored) {
+                LOGGER.warn("WebAPP:" + " APIStatusMonitor not yet deployed or" + " API :" + apiName + " not yet "
+                                    + "deployed " + " with provider: " + apiProvider);
+            }
+
+            LOGGER.info("WAIT for availability of API: " + apiName + " with version: " + apiVersion
+                                + " with provider: " + apiProvider + " with Tenant Identifier: " + tenantIdentifier
+                                + " with expected response : " + expectedResponse);
+
+            if (response != null) {
+                LOGGER.info("Data: " + response.getData());
+                if (response.getData().contains(expectedResponse)) {
+                    LOGGER.info("API :" + apiName + " with version: " + apiVersion + " with expected response "
+                                        + expectedResponse + " found");
+                    break;
+                } else {
+                    try {
+                        Thread.sleep(500);
+                    } catch (InterruptedException ignored) {
+                        // do nothing
```

Comment:
```suggestion
                        Thread.currentThread.interrupt()
```
I think we are supposed to call this here.

Reply:
+1